rustsec/src/report.rs

```diff
@@ -39,15 +43,16 @@ pub struct Report {
 }
 
 impl Report {
-    /// Generate a report for the given advisory database and lockfile
-    pub fn generate(db: &Database, lockfile: &Lockfile, settings: &Settings) -> Self {
+    /// Generate a report for the given advisory database and lockfile, optional narrowing of subgraph from target package
+    pub fn generate(db: &Database, lockfile: &Lockfile, settings: &Settings, tree:&dependency::Tree, target:Option<&Package>) -> Self {
```

Comment:
There are already two types for carrying additional state so you don't have to add parameters to this method: `Settings` and its `Settings::query` field which contains a `database::Query`.

I would think you could access the dependency tree via the `Lockfile`, and this feature could potentially set some option on the `database::Query`.

Reply:
Yes, this makes more sense. I will do a refactor using `Query`